src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/setuppayloadscanner/CHIPDeviceDetailsFragment.kt

```diff
@@ -36,48 +37,52 @@ import kotlinx.android.synthetic.main.chip_device_info_fragment.view.versionTv
-        savedInstanceState: Bundle?
-    ): View {
-        deviceInfo = checkNotNull(requireArguments().getParcelable(ARG_DEVICE_INFO))
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    deviceInfo = checkNotNull(requireArguments().getParcelable(ARG_DEVICE_INFO))
 
-        return inflater.inflate(R.layout.chip_device_info_fragment, container, false).apply {
+    return inflater.inflate(R.layout.chip_device_info_fragment, container, false).apply {
 
-            // Display CHIP setup code info to user for manual connect to soft AP
-            versionTv.text = "${deviceInfo.version}"
-            vendorIdTv.text = "${deviceInfo.vendorId}"
-            productIdTv.text = "${deviceInfo.productId}"
-            setupCodeTv.text = "${deviceInfo.setupPinCode}"
-            discriminatorTv.text = "${deviceInfo.discriminator}"
+      versionTv.text = "${deviceInfo.version}"
+      vendorIdTv.text = "${deviceInfo.vendorId}"
+      productIdTv.text = "${deviceInfo.productId}"
+      setupCodeTv.text = "${deviceInfo.setupPinCode}"
+      discriminatorTv.text = "${deviceInfo.discriminator}"
+      discoveryCapabilitiesTv.text = requireContext().getString(
+        R.string.chip_device_info_discovery_capabilities_text,
+        deviceInfo.discoveryCapabilities
+      )
 
-            if (deviceInfo.optionalQrCodeInfoMap.isEmpty()) {
-                vendorTagsLabelTv.visibility = View.GONE
-                vendorTagsContainer.visibility = View.GONE
-            } else {
-                vendorTagsLabelTv.visibility = View.VISIBLE
-                vendorTagsContainer.visibility = View.VISIBLE
+      if (deviceInfo.optionalQrCodeInfoMap.isEmpty()) {
+        vendorTagsLabelTv.visibility = View.GONE
+        vendorTagsContainer.visibility = View.GONE
+      } else {
+        vendorTagsLabelTv.visibility = View.VISIBLE
+        vendorTagsContainer.visibility = View.VISIBLE
 
-                deviceInfo.optionalQrCodeInfoMap.forEach { (_, qrCodeInfo) ->
-                    val tv = inflater.inflate(R.layout.barcode_vendor_tag, null, false) as TextView
-                    val info = "${qrCodeInfo.tag}. ${qrCodeInfo.data}, ${qrCodeInfo.intDataValue}"
-                    tv.text = info
-                    vendorTagsContainer.addView(tv)
-                }
-            }
+        deviceInfo.optionalQrCodeInfoMap.forEach { (_, qrCodeInfo) ->
+          val tv = inflater.inflate(R.layout.barcode_vendor_tag, null, false) as TextView
+          val info = "${qrCodeInfo.tag}. ${qrCodeInfo.data}, ${qrCodeInfo.intDataValue}"
+          tv.text = info
+          vendorTagsContainer.addView(tv)
         }
+      }
     }
+  }
 
-    companion object {
-        private const val ARG_DEVICE_INFO = "device_info"
+  companion object {
+    private const val ARG_DEVICE_INFO = "device_info"
 
-        @JvmStatic fun newInstance(deviceInfo: CHIPDeviceInfo): CHIPDeviceDetailsFragment {
```

Comment:
It might be a good idea reformat all the remaining Kotlin code in a separate PR so that subsequent PRs only contain relevant changes. Also, is the style part of the Android Studio project configuration? It's a pity that the restyler doesn't support Kotlin and we can't set on a single formatting style.

Reply:
Yeah, this is just the Android Studio formatting (the 4-space indentation was bothering me). 

I can do a big PR to reformat all the code, probably using [ktfmt](https://github.com/facebookincubator/ktfmt). If I have some time, I'll try https://github.com/restyled-io/restyled.io/wiki/Adding-a-Restyler.

Reply:
#8223 8223